File: ASP.NET Core Web API/Calendar.Server/Calendar.Server/Models/Event.cs
using System;

namespace Calendar.Server.Models
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime DateEnd { get; set; }
        public int TypeId { get; set; }
        public string Location { get; set; }
        public string Description { get; set; }
        public int UserId { get; set; }
        public EventType EventType { get; set; }
        public User User { get; set; }
    }
}

File: ASP.NET Core Web API/Calendar.Client/src/app/models/event.ts
export class Event
{
    id: number;
    name: string;
    dateStart: Date;
    dateEnd: Date;
    location: string;
    description: string;
    userId: number;

    constructor(id?: number, name?: string, dateStart?: Date, dateEnd?: Date, location?: string, description?: string)
    {
        this.id = id;
        this.name = name;
        this.dateStart = dateStart;
        this.dateEnd = dateEnd;
        this.location = location;
        this.description = description;
    }
}